hydrolib/hydromt_delft3dfm/dflowfm.py

```diff
@@ -0,0 +1,1096 @@
+            GeoDataFrame with the new branches to add.
+        spacing : pd.DataFrame
+            DatFrame containing spacing values per 'branchType', 'shape', 'width' or 'diameter'.
+
+        """
+        if gdf_br.crs.is_geographic:  # needed for length and splitting
+            gdf_br = gdf_br.to_crs(3857)
+
+        self.logger.info("Adding/Filling branches attributes values")
+        gdf_br = update_data_columns_attributes(gdf_br, defaults, brtype=br_type)
+
+        # If specific spacing info from spacing_fn, update spacing attribute
+        if spacing is not None:
+            self.logger.info(f"Updating spacing attributes")
+            gdf_br = update_data_columns_attribute_from_query(
+                gdf_br, spacing, attribute_name="spacing"
+            )
+
+        self.logger.info(f"Processing branches")
+        branches, branches_nodes = process_branches(
+            gdf_br,
+            branch_nodes=None,
+            id_col="branchId",
+            snap_offset=snap_offset,
+            allow_intersection_snapping=allow_intersection_snapping,
+            logger=self.logger,
+        )
+
+        self.logger.info(f"Validating branches")
+        validate_branches(branches)
+
+        # convert to model crs
+        branches = branches.to_crs(self.crs)
+        branches_nodes = branches_nodes.to_crs(self.crs)
+
+        return branches, branches_nodes
+
+    def setup_channels(
+        self,
+        channels_fn: str,
+        channels_defaults_fn: str = None,
+        spacing_fn: str = None,
+        snap_offset: float = 0.0,
+        allow_intersection_snapping: bool = True,
+    ):
+        """This component prepares the 1D channels and adds to branches 1D network
+
+        Adds model layers:
+
+        * **channels** geom: 1D channels vector
+        * **branches** geom: 1D branches vector
+
+        Parameters
+        ----------
+        channels_fn : str
+            Name of data source for branches parameters, see data/data_sources.yml.
+
+            * Required variables: [branchId, branchType] # TODO: now still requires some cross section stuff
+
+            * Optional variables: [spacing, material, shape, diameter, width, t_width, t_width_up, width_up,
+              width_dn, t_width_dn, height, height_up, height_dn, inlev_up, inlev_dn, bedlev_up, bedlev_dn,
+              closed, manhole_up, manhole_dn]
+        channels_defaults_fn : str Path
+            Path to a csv file containing all defaults values per 'branchType'.
+        spacing : str Path
+            Path to a csv file containing spacing values per 'branchType', 'shape', 'width' or 'diameter'.
+
+        """
+        self.logger.info(f"Preparing 1D channels.")
+
+        # Read the channels data
+        id_col = "branchId"
+        gdf_ch = self.data_catalog.get_geodataframe(
+            channels_fn, geom=self.region, buffer=10, predicate="contains"
+        )
+        gdf_ch.index = gdf_ch[id_col]
+        gdf_ch.index.name = id_col
```

Comment:
In the hydrolib-hydromt it seems we always need to set the index after retrieving the data frame. So this could perhaps be part of the `DataCatalog.get_geodataframe` functionality, by passing an extra parameter  e.g. `index_name` to the function.